File: LeetCode75_Artur/Array and String/1431._Kids_With_the_Greatest_Number_of_Candies.cpp
#include <iostream>
#include "vector"

using namespace std;

class Solution
{
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies)
    {
        vector<bool> res = vector<bool>(candies.size(), false);
        int max = 0;
        for(int candy : candies)
            if (candy > max)
                max = candy;

        for(size_t i = 0; i < candies.size(); ++i)
            if(candies[i] + extraCandies >= max)
                res[i] = true;

        return res;
    }
};

File: LeetCode75_Artur/Array and String/1431._Kids_With_the_Greatest_Number_of_Candies.py
from typing import List

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        flag = max(candies)
        for i in range(len(candies)):
            if candies[i] + extraCandies >= flag:
                candies[i] = True
            else:
                candies[i] = False
        return candies
    
print(Solution.kidsWithCandies(1, candies = [12,1,12], extraCandies = 10))